operator/src/main/java/oracle/kubernetes/operator/helpers/ServiceHelper.java

```diff
@@ -527,10 +534,29 @@ private Step createNewService(Step next) {
     protected abstract String getServiceCreatedMessageKey();
 
     private Step deleteAndReplaceService(Step next) {
-      V1DeleteOptions deleteOptions = new V1DeleteOptions();
+      if (getSpecType().equals(NODE_PORT_TYPE)) {
```

Comment:
Since we are replacing external type service, should we be checking serviceType == EXTERNAL here?

Reply:
I thought they the same thing since only external services are of type `NodePort`. I'll go ahead and make the change.